### ISSUE TITLE
Provide a way to disable {searchTerms} escaping on a search engine basis

### ISSUE THREAD

**Issue by LegitLlama** [CanisLupus/swift-selection-search#192]
Hi,

As per [this comment](https://github.com/CanisLupus/swift-selection-search/issues/85#issuecomment-622570654), it seems that there is a use for \*not\* escaping ```{searchTerms}``` with some search engines. An example of such an engine is the Wayback Machine, which will not support a network request such as ```https://web.archive.org/web/*/https%3A%2F%2Fgithub.com%2F``` and really need ```https://web.archive.org/web/*/https://github.com/``` in order to work.

Some use cases involve selecting text that happens to be a valid URL, or pasting a valid URL in SSS's pop-up text field, in order to search for it.

The feature you seem to have in mind to answer this is to give the user a way to disable escaping, at least for ```{seachTerms}```, on a search engine basis. Maybe you were thinking of an UI option, like the current text encoding drop down menu, or maybe of new syntax, like ```{searchTerms(noEscape)}```, or maybe something else entirely.

Thanks for your time!

**Comment by CanisLupus** [comment]
Thanks a lot for detailing this in a new issue. :) Indeed, I was thinking of a setting similar to the text encoding dropdown, but configuring searchTerms probably makes more sense! It depends a bit on the added confusion it may cause with the already existing searchTerms modifications.

**Comment by CanisLupus** [comment]
As you may have seen, I added this feature for SSS version 3.43.0 (now 3.43.2 after the short lived apocalypse from yesterday). I followed your suggestion and used {searchTerms(disableURIEncoding)} to get the desired effect (it should work on all search variables like href, etc, not only searchTerms). :) It seemed like the best approach.

Please let me know if anything is wrong.

Cheers!
Daniel

**Comment by LegitLlama** [comment]
Well, that was fast! Thanks! I'm considering using it with some regular search engines with which I would like to hide that my request is coming from an unusual place. It's going to depend on whether "usual places" inject % characters all over the place :)

For now it works just fine with the Wayback Machine!

**Comment by CanisLupus** [comment]
Great! :) It was much faster than my usual, yes. 😅 I figured this was simple enough and didn't have conflicts of note with all the other options.

**Comment by ZIEXED** [comment]
Applied to Every engine in my list, I'm wondering why it's being URI Encoded in the first place

**Comment by CanisLupus** [comment]
@ZIEXED now I'm curious... Why did you add disableURIEncoding to all your engines? Were you encountering problems with some websites like the Internet Archive? I wouldn't recommend doing this for all of them.

It's encoded by default because it is what most websites do when you write on their search bars. Essentially to sanitize user input. If you search for something that includes characters like `:`, `/`, `?`, etc (which can be parts of a URL) on Google, IMDB, etc, you will notice that it encodes those, just like SSS.

Example: If you disable encoding for Google Maps and search for `some place / another` WITHOUT encoding, your search will become just `some place ` since the website will consider the slash as part of the URL path.

**Comment by ZIEXED** [comment]
LegitLlama comment above made me curious so I checked the sites in my list to see if they do URI Encodeing but I was just using spaces to see if they will be encoded but none of them encoded the space so I assumed that my requests is known to be coming from an irregular place, and  it didn't cross my mind the problem with :, /, ?,  characters
reverted back :O

**Comment by CanisLupus** [comment]
I see. :) Btw, spaces are sometimes encoded as `+` (ex.: Google), but `%20` is the proper encoding.

**Comment by LegitLlama** [comment]
@ZIEXED Sorry that I confused you, I should have been clearer, especially considering that it's URL encoding is a security feature, that I didn't expect many of my search engines not to use it. Only the most unusual ones. For example I checked DeepL, it uses a POST request with JSON in which the user input is not sanitized client-side. So I disabled encoding there, after double checking that encoding leads to inaccurate translations.

Neither SSS nor Firefox `about:preferences#search` will be able to reproduce DeepL's "natural" request though, so I can't hide that my request is unnatural, and I'm probably one of few people to use it.

I only disabled encoding for DeepL and the Wayback Machine so far.
